src/ampal/analyse_protein.py

```diff
@@ -127,7 +127,32 @@ def sequence_isoelectric_point(seq, granularity=0.1):
     return ph_range[pi_index]
 
 
-def measure_sidechain_torsion_angles(residue, verbose=True):
+def classify_angle_as_rotamer(angle: float) -> int:
+    """Classifies a chi angle into a rotamer.
+
+    Notes
+    -----
+    This function uses the convention 1:g+ 2:t 3:g- as from 10.1006/jmbi.1993.1170
+    R L Dunbrack Jr, M Karplus (1993) Backbone-dependent rotamer library for proteins.
+    Application to side-chain prediction.
+
+    Parameters
+    ----------
+    angle : float
+        Chi angle to classify.
+    """
+    if 0 < angle < 120:
+        return 1  # g+
+    elif -120 < angle < 0:
+        return 3  # g-
+    elif -120 > angle > -180 or 120 < angle < 180:
+        return 2  # t
+    else:
+        print(f"Got angle {angle} which does not between 0 < x < 180 or -180 > x > 0 ")
+        return None
+
+
+def measure_sidechain_torsion_angles(residue, verbose=True, return_rotamers=False):
```

Comment:
Could you add a type annotation here? To make the return type monomorphic you could return `Tuple[List[float], Optional[List[int]]]`.